### PR TITLE
vectorstores: Add azureaisearch

### PR DESCRIPTION
**Pull request by ThomasP1988**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Comment by ThomasP1988** [comment]
needs to be tested with environment variables: AZURE_AI_SEARCH_ENDPOINT, AZURE_AI_SEARCH_API_KEY and OPENAI_API_KEY

**Comment by tmc** [comment]
Can you set those tests to t.Skip if the necessary environment variables are not set?

**Comment by tmc** [comment]
You can pass a skip reason to Skip()

**Comment by tmc** [comment]
can you rebase this to main?

**Comment by tmc** [comment]
See comments on https://github.com/tmc/langchaingo/pull/464/files#r1437955293

File: vectorstores/azureaisearch/azureaisearch.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+type Store struct {
```
Comment (tmc) [inline vectorstores/azureaisearch/azureaisearch.go:16]: please attach comments to exported symbols

File: vectorstores/azureaisearch/azureaisearch.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		}
+		ids = append(ids, id)
+	}
+
+	return ids, nil
+}
+
+// SimilaritySearch creates a vector embedding from the query using the embedder
+// and queries to find the most similar documents.
+func (s Store) SimilaritySearch(
```
Comment (tmc) [inline vectorstores/azureaisearch/azureaisearch.go:97]: Can you make the receiver a pointer for consistency?

Comment (ThomasP1988) [reply vectorstores/azureaisearch/azureaisearch.go:97]: sorry, can you be more specific?

Comment (tmc) [reply vectorstores/azureaisearch/azureaisearch.go:97]: (s *Store) instead of (s Store)

**Comment by tmc** [comment]
Thanks, can you follow on and add an example of using this and the opensearch implementation?

**Pull request merged**